pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/ConfigDrivenServiceProviderBase.java

```diff
@@ -77,100 +65,51 @@
      *
-     * @return the special comparator for ordering config bean instance ids
+     * @param instanceId of this provider, root provider is hardcoded to {@code root}, managed instances are named based on
+     *                  the config bean name
      */
-    static Comparator<String> configBeanComparator() {
-        return BEAN_INSTANCE_ID_COMPARATOR;
-    }
-
-    static BindableConfigBeanRegistry resolveConfigBeanRegistry() {
-        HelidonConfigBeanRegistry cbr = ConfigBeanRegistryHolder.configBeanRegistry().orElse(null);
-        if (cbr == null) {
-            LOGGER.log(System.Logger.Level.INFO, "Config-Driven Services is disabled (config bean registry not found)");
-            return null;
-        }
-
-        if (!(cbr instanceof BindableConfigBeanRegistry)) {
-            Optional<CallingContext> callingContext = CallingContextFactory.create(false);
-            String desc = "Config-Driven Services is disabled (unsupported implementation): " + cbr;
-            String msg = (callingContext.isEmpty()) ? toErrorMessage(desc) : toErrorMessage(callingContext.get(), desc);
-            throw new PicoException(msg);
-        }
-
-        return (BindableConfigBeanRegistry) cbr;
-    }
-
-    /**
-     * The map of bean instance id's to config bean instances.
-     *
-     * @return map of bean instance id's to config bean instances
-     */
-    public Map<String, Object> configBeanMap() {
-        return Map.copyOf(configBeanMap);
-    }
-
-    /**
-     * The map of bean instance id's to managed configured service providers that this instance managed directly.
-     *
-     * @return map of bean instance id to managed configured service providers
-     */
-    public Map<String, Optional<AbstractConfiguredServiceProvider<T, CB>>> configuredServicesMap() {
-        return Collections.unmodifiableMap(managedConfiguredServicesMap);
+    protected ConfigDrivenServiceProviderBase(String instanceId) {
+        this.instanceId = instanceId;
     }
 
     /**
      * Called during initialization to register a loaded config bean.
      *
-     * @param instanceId the config bean instance id
      * @param configBean the config bean
      */
-    public void registerConfigBean(String instanceId,
-                                   Object configBean) {
-        Objects.requireNonNull(instanceId);
+    @Override
+    public final void registerConfigBean(NamedInstance<CB> configBean) {
         Objects.requireNonNull(configBean);
         assertIsInitializing();
 
-        Object prev = configBeanMap.put(instanceId, configBean);
-        assert (prev == null);
-
-        prev = managedConfiguredServicesMap.put(instanceId, Optional.empty());
+        Object prev = managedConfiguredServicesMap.put(configBean.name(), createInstance(configBean));
         assert (prev == null);
     }
 
     @Override
     public boolean reset(boolean deep) {
         super.reset(deep);
-        configBeanMap.clear();
```

Comment:
why was this removed?

Reply:
Because the config bean map is not needed - we have a map of managed service providers, which have a reference to config beans that created them. So this map was redundant.